### DISCUSSION TITLE
`(str, Enum)` fields are no longer (automatically) treated as strings in SQL

### DISCUSSION THREAD

**Discussion by nayaverdier** [fastapi/sqlmodel#717] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum
from sqlmodel import SQLModel, Field

class SomeEnum(str, Enum):
    FOO = "FOO"
    BAR = "BAR"

class SomeModel(SQLModel, table=True):
    some_enum: SomeEnum = Field(primary_key=True)

# `SomeModel.some_enum.type` is the following
# Enum('FOO', 'BAR', name='someenum')

# in previous versions, the type was:
# AutoString()
```

### Description

Prior to `0.0.9`, all of my `str, Enum` types were utilizing `AutoString` and stored as varchars in postgres. Since #669, these types are now converted to postgres Enum fields, which is not desirable for my use case.

A very hacky workaround I found is to monkeypatch `sa_Enum` in the `sqlmodel.main` module, which works since I am not using sqlalchemy.Enum fields anywhere:
```python
from sqlmodel import main as _sqlmodel_main

_sqlmodel_main.sa_Enum = lambda _: _sqlmodel_main.AutoString  # type: ignore
```

I know another solution is to override every enum field with a `sqlmodel.Field(sa_column=...)`, but I'd prefer a global solution to treat all `(str, Enum)` types as `AutoString` instead of `sqlalchemy.Enum`.

### Operating System

Linux

### SQLModel Version

0.0.12

### Python Version

3.11.5

**Comment by antont** [comment]
Could there maybe be a generic solution for overriding the default mappings from python types to sql?

BTW we were using postgres string enums, and maybe it's a good default, but I figure not using them is a valid usecase too.

**Comment by emilhdiaz** [comment]
This change in the handling of Enums is a breaking change for existing schemas that previously used the VARCHAR field to store python Enums. The previous values (the python Enum **values**) aren't compatible with the new database Enum values (derived from the python Enum **names**). This forces projects to add additional migration steps to map the existing values or modify code to use the `ChoiceType` from `sqlalchemy_utils`. 

Would love to see a simple (and supported) way to change default mappings

**Comment by gvoysey** [comment]
Just got nailed by this on an upgrade branch (brought about by other dependencies forcing pydantic v2).

Any resolution or tips/tricks in the past 10 months?

**Comment by brendanator** [comment]
Here's an alternative that stores the str values in the database and uses the StrEnum values in Python

```python
from sqlmodel import main as _sqlmodel_main

# Patch sa_Enum to use the enum string values rather than the enum member names in the database
def _parse_into_values(self, enums, kw):
    if not enums and "_enums" in kw:
        enums = kw.pop("_enums")
    if len(enums) == 1 and hasattr(enums[0], "__members__"):
        self.enum_class = enums[0]
        # Use the str values for the database column
        values = [e.value for e in self.enum_class]
        # Use the enum members for the Python objects
        objects = list(self.enum_class)
        return values, objects
    else:
        self.enum_class = None
        return enums, enums


_sqlmodel_main.sa_Enum._parse_into_values = _parse_into_values  # type: ignore
```
